test/k8s-integration/cluster.go

```diff
@@ -269,31 +276,103 @@ func clusterUpGKE(gceZone, gceRegion string, numNodes int, imageType string) err
 		}
 	}
 
-	var cmd *exec.Cmd
-	cmdParams := []string{"container", "clusters", "create", *gkeTestClusterName,
-		locationArg, locationVal, "--num-nodes", strconv.Itoa(numNodes),
-		"--quiet", "--machine-type", "n1-standard-2", "--image-type", imageType}
-	if isVariableSet(gkeClusterVer) {
-		cmdParams = append(cmdParams, "--cluster-version", *gkeClusterVer)
+	if useManagedDriver {
+		accessToken, err := exec.Command("gcloud", "auth", "print-access-token").Output()
+		if err != nil {
+			return fmt.Errorf("failed to get authentication token: %v", err)
+		}
+
+		token := &oauth2.Token{AccessToken: string(accessToken[:len(accessToken)-1])}
+		oauthClient := oauth2.NewClient(context.Background(), oauth2.StaticTokenSource(token))
+		service, err := container.NewService(context.Background(), option.WithHTTPClient(oauthClient))
+		if err != nil {
+			return fmt.Errorf("failed to create a new service: %v", err)
+		}
+
+		service.BasePath = "https://test-container.sandbox.googleapis.com/" // e.g test, staging env URL
+
+		request := &container.CreateClusterRequest{
+			Cluster: &container.Cluster{
+				Name:             *gkeTestClusterName,
+				InitialNodeCount: int64(numNodes),
+				// InitialClusterVersion: "latest",
+				AddonsConfig: &container.AddonsConfig{
+					GcpFilestoreCsiDriverConfig: &container.GcpFilestoreCsiDriverConfig{Enabled: true},
+				},
+				ReleaseChannel: &container.ReleaseChannel{Channel: "RAPID"},
+				NodeConfig: &container.NodeConfig{
+					MachineType: "n1-standard-2",
+					ImageType:   imageType,
+				},
+			},
+		}
+
+		klog.Infof("Creating kubernetes e2e cluster on gke")
+		op, err := service.Projects.Locations.Clusters.Create("projects/leiyiz-gke-dev/locations/us-central1-b", request).Do()
```

Comment:
It looks like the boskos project gets put into gcloud config, so you could just get the current project (and location?) from gcloud config get-value like is done below.

Reply:
done. refactored how it was done